### ISSUE TITLE
Optimize scrolling the trades page by checking Member Points

### ISSUE THREAD

**Issue by tomreece** [tomreece/pucauto#36]
Idea submitted by a user:

Sort the trades page by Member Points and scroll down just far enough until you reach a member with points less than `min_value`. If their points are < `min_value` we won't make a trade to them anyway. This could speed up the loading step.

Try it out and monitor performance gains first.

**Comment by tomreece** [comment]
I played around with something like this tonight on a large trade list and the performance gains were incredible. So much time was wasted loading traders with low point values. 5 seconds to reach a bottom of 97 points, which is above my min of 150. 40 seconds to reach a bottom of 15 points. 35 seconds of wasted time eliminated because I'm not going to be trading with those low point members anyway. Need to get this into master ASAP:

```
-def load_full_trade_list():
+def load_full_trade_list(partial=False):
     """Scroll to the bottom of the page until we can't scroll any further. PucaTrade's /trades page implements an
     infinite scroll table. Without this function, we could only see a portion of the cards available for trade."""

@@ -171,7 +176,14 @@ def load_full_trade_list():
         DRIVER.execute_script("window.scrollBy(0, 5000);")
         wait_for_load()
         new_scroll_y = DRIVER.execute_script("return window.scrollY;")
-        if new_scroll_y == old_scroll_y or new_scroll_y < old_scroll_y:
+        try:
+            lowest_points_in_table = int(DRIVER.find_element_by_css_selector(".cards-show tbody tr:last-of-type td.points").text)
+        except:
+            lowest_points_in_table = 0
+            pass
+        print(lowest_points_in_table)
+        if new_scroll_y == old_scroll_y or new_scroll_y < old_scroll_y or (partial and lowest_points_in_table < CONFIG.get("min_value")):
+            print(new_scroll_y)
             break
         else:
             old_scroll_y = new_scroll_y
```

PS: Remember to do a one-time sort by user_points at startup.

```
+    DRIVER.find_element_by_css_selector("th[title='user_points']").click()
+    wait_for_load()
```

**Comment by tomreece** [comment]
Something to consider: Should I do this for add on searching too or will we potentially be missing out on users to add on to?

**Comment by isutrumpet** [comment]
The risk when building the dictionary of missing add-ons but drastically increasing performance is probably worth it to any user with a large send list.  If there is a way to split the functionality (maybe only do a "full add-on check" every 30 minutes or so?) that would be ideal.  Getting down the trade search time is extremely important when scrolling down is taking too long, however.

**Comment by tomreece** [comment]
@isutrumpet good point and idea. Search for addons less frequently. I can make this stuff configurable but I hate software that is TOO configurable so I'm trying to strike a balance between what needs to be configured and what I think is best. I will implement something here soon.

**Comment by tomreece** [comment]
Fixed in #41.

@isutrumpet I also added configuration to check for add ons every x minutes that you can set in the config.json, check out the patch notes here: https://github.com/tomreece/pucauto/releases/tag/v0.4.1
